src/components/Header/Header.tsx

```diff
@@ -0,0 +1,118 @@
+import React, { useState, useEffect } from 'react';
+import cn from 'classnames';
+
+import { Todo } from '../../types/Todo';
+import { USER_ID } from '../../api/todos';
+import * as todoService from '../../api/todos';
+import { showErrorMesage } from '../../utils/showErrorMesage';
+
+type Props = {
+  todos: Todo[];
+  titleText: string;
+  titleFunction: (el: string) => void;
+  errorFunction: (el: string) => void;
+  todosFunction: (el: Todo[] | ((currentTodos: Todo[]) => Todo[])) => void;
+  loadingTodoFunction: (el: Todo | null) => void;
+  inputRef: React.RefObject<HTMLInputElement>;
+  isDeleting: boolean;
+  focusInput: () => void;
+};
+
+export const Header: React.FC<Props> = ({
+  todos,
+  titleText,
+  titleFunction,
+  errorFunction,
+  todosFunction,
+  loadingTodoFunction,
+  inputRef,
+  isDeleting,
+  focusInput,
+}) => {
+  const [loading, setLoading] = useState(false);
+
+  //#region Setvice Functions
+  function addTodos(title: string, completed: boolean, userId: number) {
```

Comment:
you can just move this function to App.tsx and pass to Header like a prop, after that you can remove a lot of props in Header